### ISSUE TITLE
1.6.3 fails to install deps with ruby 3.1.3

### ISSUE THREAD

**Issue by modosc** [ruby/debug#852]
**Your environment**

* `ruby -v`:  `ruby 3.1.3p185 (2022-11-24 revision 1a6b16756e) [x86_64-darwin22]` and `ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin21]`
* `rdbg -v`: `rdbg 1.6.3` and `rdbg 1.6.2`
* `bundler --version`: `Bundler version 2.3.21`
 
**Describe the bug**
when debug-1.6.3 is installed with ruby-3.1.3 it leaves out all of its deps

**To Reproduce**
given this:

`Gemfile`:
```
source 'https://rubygems.org'
gem 'debug'
```
`.ruby-version`
```
3.1.2
```

`Gemfile.lock`:
```
GEM
  remote: https://rubygems.org/
  specs:
    debug (1.6.3)
      irb (>= 1.3.6)
      reline (>= 0.3.1)
    io-console (0.5.11)
    irb (1.5.1)
      reline (>= 0.3.0)
    reline (0.3.1)
      io-console (~> 0.5)

PLATFORMS
  x86_64-darwin-21
  x86_64-darwin-22

DEPENDENCIES
  debug

BUNDLED WITH
   2.3.21
```

when i change `.ruby-version` to `3.1.3` and run `bundle update` all of the `debug` deps are removed:
```diff
diff --git a/Gemfile.lock b/Gemfile.lock
index 856ddda..01c43e4 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -2,13 +2,6 @@ GEM
   remote: https://rubygems.org/
   specs:
     debug (1.6.3)
-      irb (>= 1.3.6)
-      reline (>= 0.3.1)
-    io-console (0.5.11)
-    irb (1.5.1)
-      reline (>= 0.3.0)
-    reline (0.3.1)
-      io-console (~> 0.5)
```

this then causes issues:
```
Downloading debug-1.6.3 revealed dependencies not in the API or the lockfile                                                                   
(irb (>= 1.3.6), reline (>= 0.3.1)).                                                                                                                                      
Either installing with `--full-index` or running `bundle update debug` should                                                                                             
fix the problem.                              
```

and neither of the solutions mentioned solved the problem.                                                                                                                            
**Expected behavior**
this shouldn't happen

**Additional context**
verified on macos and linux

**Comment by ko1** [comment]
I heard that Ruby 3.1.3 has a packaging issue.
Could you try `gem install debug` or can't do that with same version?
I'll release debug 1.7.0 soon and try it later.

**Comment by ko1** [comment]
could you try to install 1.7.0?

**Comment by urmora** [comment]
> 

For me, 1.7.1 fails (same way, inside CI pipeline, release version of ruby 3.2.0) even if I mention missing gems in Gemfile. 1.7.0 seems to work.

**Comment by lxxxvi** [comment]
Same issue here with `1.7.1`, `bundle install` fails in GitHub Action with this message:

```
Downloading debug-1.7.1 revealed dependencies not in the API or the lockfile
  (irb (>= 1.5.0), reline (>= 0.3.1)).
  Either installing with `--full-index` or running `bundle update debug` should
  fix the problem.
```

These dependencies disappear when upgrading from `1.7.0`.

(Ruby `3.2.0`,  Bundler `2.4.2`)

**Comment by sudara** [comment]
I'm currently seeing this on 1.7.1 with Ruby 3.2.0 and Bundler 2.3.5

https://github.com/sudara/alonetone/actions/runs/3839336156/jobs/6536959125

**Comment by msducheminjr** [comment]
I actually got my GitHub action to work on `1.6.3` when it was failing on a `1.7.1` bundle update.

https://github.com/statelesscode/nerd_dice_dot_com/actions/runs/4029577417
https://github.com/statelesscode/nerd_dice_dot_com/actions/runs/4029686179

When I tried to do a `bundle update` and even a `bundle update --full-index` the `debug` gem would not write the `irb` and `reline` dependencies to the `Gemfile.lock`. (As evidenced in [ed62fc20](https://github.com/statelesscode/nerd_dice_dot_com/commit/ed62fc20db069ad4c9075c2b87b6450c5c5cad0d)).

The bundle was updated on Ruby `3.2.0` installed with `rvm` on an Ubuntu 22.04 system.  Putting a pessimistic constraint on `debug` in commit [e0270226d](https://github.com/statelesscode/nerd_dice_dot_com/commit/e0270226d704faa315f6b3f3acc398130960a7ff) and re-running `bundle update` caused the `Gemfile.lock` to write correctly which made the build pass.

**Comment by davekaro** [comment]
I had this same issue, but it didn't manifest until I went from Ruby 3.2.0/Bundler 2.4.3 to Ruby 3.2.1/Bundler 2.4.6. Suddenly Github actions failed with:

> Downloading debug-1.7.1 revealed dependencies not in the API or the lockfile
  (irb (>= 1.5.0), reline (>= 0.3.1)).

I fixed for now by adding those gems manually to my Gemfile. Not sure where to look next as those depencies are listed in the gemspec.

Also, I looked up the history of my Gemfile.lock and back on January 4 I ran a `bundle update` that included this change:
```
     debug (1.7.1)
-      irb (>= 1.5.0)
-      reline (>= 0.3.1)
``` 

I remember when that happened and didn't understand why since nothing really changed as far as I remember.

**Comment by benoittgt** [comment]
I had the same issue.
I do not think it is related to the version itself but more of an issue on resolving between how `irb` and `reline` are installed (native gem VS gem handled by Gemfile). 

```sh
$ gem uninstall debug irb reline
```
Clean mentions of "irb" and "reline" in Gemfile.lock. You should have only "debug (1.7.1)" line.
```sh
$ bundle update debug
```

Check `Gemfile.lock`. You should see a proper output:
```
    debug (1.7.1)
      irb (>= 1.5.0)
      reline (>= 0.3.1)
```

**Comment by miharekar** [comment]
Same happening to me on 3.2.1. I thought it was fixed with https://github.com/rubygems/rubygems/issues/6082 but it came back.

The way I fixed it was to comment out `gem "debug"` from Gemfile, then `bundle`, then `gem install debug`, then uncommenting and one last `bundle`.

**Comment by elalemanyo** [comment]
We are running Ruby 3.2.1 and none of the solutions help. In the end, downgrading to 1.6.3 was the answer.

**Comment by mdoyle13** [comment]
Starting getting failures on GithubActions today. Our project is running Rails7 and Ruby 3.2.0.
```
Downloading debug-1.7.1 revealed dependencies not in the API or the lockfile
(irb (>= 1.5.0), reline (>= 0.3.1)).
```

Locking version to 1.7.0 resolves the GitHub Actions issue for me
```
gem 'debug', '= 1.7.0'
```

**Comment by searls** [comment]
For anyone encountering this issue again on 3.2.1 and Bundler 2.4.7, I'm also experiencing it in GitHub actions:

> Downloading debug-1.7.1 revealed dependencies not in the API or the lockfile (irb (>= 1.5.0), reline (>= 0.3.1)). Either installing with `--full-index` or running `bundle update debug` should fix the problem.

When I look on disk, the built-in debug-1.7.1.gemspec indeed declares no runtime dependencies and as a result my Gemfile.lock also resolved no transitive dependencies on `irb` or `reline`:

```
    # …
    date (3.3.3)
    debug (1.7.1)
    discard (1.2.1)
      activerecord (>= 4.2, < 8)
    # …
```

@deivid-rodriguez's workaround from https://github.com/rubygems/rubygems/issues/6082 still works, though:

1. Run `gem install debug`
2. Comment out `gem "debug"` in the Gemfile
3. `bundle install` will remove it from the lockfile
4. Uncomment `gem "debug"`
5. `bundle install` will add the transitive deps to the lockfile

```diff
     date (3.3.3)
     debug (1.7.1)
+      irb (>= 1.5.0)
+      reline (>= 0.3.1)
     i18n (1.12.0)
       concurrent-ruby (~> 1.0)
+    io-console (0.6.0)
+    irb (1.6.2)
+      reline (>= 0.3.0)
     rake (13.0.6)
     regexp_parser (2.7.0)
+    reline (0.3.2)
+      io-console (~> 0.5)
     request_store (1.5.1)
```

This was the fewest steps I could figure out to get things working again.

**Comment by hsbt** [comment]
see https://github.com/ruby/debug/issues/898#issuecomment-1432741515

**Comment by melvinotieno** [comment]
> For anyone encountering this issue again on 3.2.1 and Bundler 2.4.7, I'm also experiencing it in GitHub actions:
> 
> > Downloading debug-1.7.1 revealed dependencies not in the API or the lockfile (irb (>= 1.5.0), reline (>= 0.3.1)). Either installing with `--full-index` or running `bundle update debug` should fix the problem.
> 
> When I look on disk, the built-in debug-1.7.1.gemspec indeed declares no runtime dependencies and as a result my Gemfile.lock also resolved no transitive dependencies on `irb` or `reline`:
> 
> ```
>     # …
>     date (3.3.3)
>     debug (1.7.1)
>     discard (1.2.1)
>       activerecord (>= 4.2, < 8)
>     # …
> ```
> 
> @deivid-rodriguez's workaround from [rubygems/rubygems#6082](https://github.com/rubygems/rubygems/issues/6082) still works, though:
> 
> 1. Run `gem install debug` to fix the incorrect gemspec on disk
> 2. Comment out `gem "debug"` in the Gemfile
> 3. `bundle install` will remove it from the lockfile
> 4. Uncomment `gem "debug"`
> 5. `bundle install` will add the transitive deps to the lockfile
> 
> ```diff
>      date (3.3.3)
>      debug (1.7.1)
> +      irb (>= 1.5.0)
> +      reline (>= 0.3.1)
>      i18n (1.12.0)
>        concurrent-ruby (~> 1.0)
> +    io-console (0.6.0)
> +    irb (1.6.2)
> +      reline (>= 0.3.0)
>      rake (13.0.6)
>      regexp_parser (2.7.0)
> +    reline (0.3.2)
> +      io-console (~> 0.5)
>      request_store (1.5.1)
> ```
> 
> This was the fewest steps I could figure out to get things working again.

Unfortunately doesn't work for me

**Comment by melvinotieno** [comment]
Seems like version 1.7.0 of debug works fine so I have had to pin it to that version to work

**Comment by msducheminjr** [comment]
I temporarily solved for it by just adding the dependencies back into the Gemfile.lock. It will kick again the next time you do a `bundle update`, so it's not sustainable. It works as a temporary stopgap to use `1.7.1` and have your GitHub action builds succeed.

[Upgrade to Ruby 3.2.1 and Rails 7.0.4.3 section on debug gem](https://www.youtube.com/watch?v=XvRzpoWfM9Q&t=789s)

**Comment by hsbt** [comment]
This issue has been fixed with Ruby 3.2.2 and 3.1.4.
